bcda/cli.go

```diff
@@ -156,26 +229,41 @@ func importCCLFDirectory(filePath string) (success, failure, skipped int, err er
 	if err != nil {
 		return 0, 0, 0, err
 	}
-	for _, file := range cclf8 {
-		//todo match cclf0 and validate
-		err = importCCLF8(file)
+
+	for _, file := range cclf0 {
+		aco := file.acoID
+		cclfValidator, err := importCCLF0(file)
 		if err != nil {
-			failure += 1
-			continue
+			log.Error(err)
+			return 0, 0, 0, err
 		}
-		log.Info(fmt.Sprintf("Successfully imported CCLF8 file: %v", file))
-		success += 1
-	}
-	for _, file := range cclf9 {
-		//todo match cclf0 and validate
-		err = importCCLF9(file)
-		if err != nil {
-			failure += 1
-			continue
+
+		for _, file = range cclf8 {
+			if file.acoID == aco {
+				err = importCCLF8(file, cclfValidator)
+				if err != nil {
+					log.Error(err)
+					failure++
+					continue
+				}
+				log.Info(fmt.Sprintf("Successfully imported CCLF8 file: %v", file))
+				success++
+			}
+		}
+		for _, file = range cclf9 {
+			if file.acoID == aco {
+				err = importCCLF9(file, cclfValidator)
+				if err != nil {
+					log.Error(err)
+					failure++
+					continue
+				}
+				log.Info(fmt.Sprintf("Successfully imported CCLF9 file: %v", file))
+				success++
+			}
 		}
-		log.Info(fmt.Sprintf("Successfully imported CCLF9 file: %v", file))
-		success += 1
 	}
+
```

Comment:
Does it matter if a CCLF0 file fails to find the associated CCLF8/9?  Maybe we should change sortCCLFFiles(cclf0, cclf8, cclf9 *[]cclfFileMetadata) to sortCCLFFiles(*map[string][]cclfFileMetadata) to ease up on the sorting load?  I think we could get that done in a single pass instead of up to 600 passes as it is now.

Reply:
to your first question, we have to answer that as a team. And I think the way you wrote sortCCLFFiles(cclf0, cclf8, cclf9 *[]cclfFileMetadata) is fine for now, at least I didn't see a problem with it as I was coding.